src/models/enemy.js

```diff
@@ -0,0 +1,54 @@
+import Phaser from 'phaser'
+
+import {CONFIG} from '../config'
+import {WORLD} from '../world'
+import {Triangle} from './base'
+
+const enemySize = CONFIG.enemies.size
+const {Triangle: PTriangle} = Phaser.Geom
+
+const _enemy = (x, y, length, hitPoints, color) => {
+  const geom = Triangle.equilateral(x, y, length)
+  let hp = hitPoints
+  const draw = () => {
+    WORLD.graphics.lineStyle(2, CONFIG.colors.white, 2)
+    WORLD.graphics.fillStyle(color, 1)
+    WORLD.graphics.fillTriangleShape(geom)
+  }
+  const animateIdle = () => {
+    PTriangle.Rotate(geom, 0.1)
+  }
+  const applyBehavior = () => {
+    // TODO!
+    //  ie: BEHAVIOR.bobAndWeave, or BEHAVIOR.chasePlayer
+    //  * should be passed in at creation and changed based on state etc.
+    //  * like an enemy that has 'phases' triggered by hitpoints etc.
+  }
+  const update = () => {
+    animateIdle()
+    applyBehavior()
+    draw()
+  }
+  const onHit = () => hp--
+  draw()
+  return {
+    _triangle: geom,
+    properties: {
+      x: () => x,
+      y: () => y,
+      length: () => length,
+      hp: () => hp
+    },
```

Comment:
I'm strongly leaning towards just using MobX for model props / events, especially because I want things like the geometry to be computed, so I would strongly consider this pattern TBD